roles/reproducer/tasks/devscripts_layout.yml

```diff
@@ -1,6 +1,224 @@
 ---
-- name: Run devscripts role
-  vars:
-    cifmw_use_reproducer: true
+- name: Ensure we have up-to-date devscripts config
+  ansible.builtin.import_role:
+    name: "devscripts"
+    tasks_from: "build_config.yml"
+
+- name: Check if devscript cluster exists
   ansible.builtin.import_role:
-    name: devscripts
+    name: "devscripts"
+    tasks_from: "110_check_ocp.yml"
+
+- name: Push OCP related DNS entries
+  become: true
+  notify: "Restart dnsmasq"
+  vars:
+    _cluster: "{{ cifmw_devscripts_config['cluster_name'] }}"
+    _pub_net_data: >-
+      {{
+        _libvirt_manager_networking.networks[cifmw_libvirt_manager_pub_net]
+      }}
+    _api_addr: >-
+      {% if _pub_net_data.network_v4 is defined -%}
+      {{
+        _pub_net_data.network_v4 | ansible.utils.nthhost(2)
+      }}
+      {% endif -%}
+      {% if _pub_net_data.network_v6 is defined -%}
+      {{
+        _pub_net_data.network_v6 | ansible.utils.nthhost(2)
+      }}
+      {% endif -%}
+    _apps_addr: >-
+      {% if _pub_net_data.network_v4 is defined -%}
+      {{
+        _pub_net_data.network_v4 | ansible.utils.nthhost(3)
+      }}
+      {% endif -%}
+      {% if _pub_net_data.network_v6 is defined -%}
+      {{
+        _pub_net_data.network_v6 | ansible.utils.nthhost(3)
+      }}
+      {% endif -%}
+  ansible.builtin.copy:
+    dest: "/etc/cifmw-dnsmasq.d/ocp_dns.conf"
+    mode: "0644"
+    content: |
+      local=/{{ _cluster }}.openstack.lab/
+      domain={{ _cluster }}.openstack.lab
+      address=/api.{{ _cluster }}.openstack.lab/{{ _api_addr }}
+      address=/api-int.{{ _cluster }}.openstack.lab/{{ _api_addr }}
+      address=/.apps.{{ _cluster }}.openstack.lab/{{ _apps_addr }}
+      localise-queries
+    validate: "/usr/sbin/dnsmasq -C %s --test"
+
+- name: Configure NetworkManager
+  become: true
+  notify: Restart NetworkManager
+  ansible.builtin.copy:
+    dest: >-
+      /etc/NetworkManager/conf.d/{{ cifmw_reproducer_networkmanager_config }}
+    mode: "0644"
+    content: |
+      [main]
+      rc-manager=none
+      dns=none
+    validate: "NetworkManager --print-config %s"
+
+- name: Ensure services are reloaded
+  ansible.builtin.meta: flush_handlers
+
+- name: Ensure we use dnsmasq as resolver
+  become: true
+  vars:
+    _cluster: "{{ cifmw_devscripts_config['cluster_name'] }}"
+  ansible.builtin.copy:
+    dest: /etc/resolv.conf
+    content: |
+      search {{ _cluster }}.openstack.lab
+      nameserver {{ _libvirt_manager_networking.networks.ctlplane.gw_v4 }}
```

Comment:
When deploying IPv6 there is no `gw_v4`.

Reply:
we now rely on dnsmasq within networkmanager, with zone delegation to our own instance. Should be sufficient and far, far better.